### PR TITLE
Fixup shutdown room API

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/handlers/message.py [hunk 1]
```diff
@@ -392,6 +392,13 @@ def assert_accepted_privacy_policy(self, requester):
         if requester.app_service is not None:
             return
 
+        # Check if the user has accepted the privacy policy. We only do this if
```
Comment (richvdh) [inline synapse/handlers/message.py]: should we not do this by skipping the call to `assert_accepted_privacy_policy`, in the same way that we currently set `ratelimit=False` ?

File: synapse/storage/room.py [hunk 2]
```diff
@@ -500,10 +500,12 @@ def get_all_new_public_rooms(txn):
 
     @defer.inlineCallbacks
     def block_room(self, room_id, user_id):
-        yield self._simple_insert(
+        yield self._simple_upsert(
```
Comment (richvdh) [inline synapse/storage/room.py]: has this just been broken since we did the native upsert thing?

Comment (erikjohnston) [reply synapse/storage/room.py]: No? I don't really follow.

Comment (richvdh) [reply synapse/storage/room.py]: well, why are you changing it?

File: synapse/handlers/message.py [hunk 3]
```diff
@@ -388,17 +391,6 @@ def assert_accepted_privacy_policy(self, requester):
         if self._block_events_without_consent_error is None:
             return
 
-        # exempt AS users from needing consent
```
Comment (richvdh) [inline synapse/handlers/message.py]: do we not still need this? is it covered by the is_exempt clause?

File: synapse/storage/room.py (hunk 2 above)
Comment (erikjohnston) [reply synapse/storage/room.py]: So that it doesn't fail when we call it twice, as per comment in admin code

Comment (richvdh) [reply synapse/storage/room.py]: heh, I'd failed to spot that you were changing insert to upsert. makes much more sense now.

File: tests/rest/client/v1/test_admin.py [hunk 4]
```diff
@@ -366,47 +366,52 @@ def prepare(self, reactor, clock, hs):
...
+        self.other_user_token = self.login("user", "pass")
+
+        # Mark the admin user as having consented
+        self.get_success(
+            self.store.user_set_consent_version(self.admin_user, "1"),
+        )
 
-        other_user = self.register_user("user", "pass")
-        other_user_token = self.login("user", "pass")
+    def test_shutdown_room_conset(self):
```
Comment (richvdh) [inline tests/rest/client/v1/test_admin.py]: conset

Comment (richvdh) [inline tests/rest/client/v1/test_admin.py]: can you add a docstring that explains in slightly more words what you are testing for here?

**Pull request merged**